e2e/api/api.py

```diff
@@ -272,5 +276,10 @@ def __str__(self) -> str:
         return '{}({})'.format(self.__class__.__qualname__, self.url)
 
     def __repr__(self) -> str:
-        return '{}({}, timeout={})'.format(self.__fqualname__, repr(self.url),
-                                           repr(self._timeout))
+        # XXX: Sort the dictionary so we get consistent results since not all
```

Comment:
I've been using `XXX` as a way to grab attention since it's highlighted in most editors/IDEs. I can remove it if pylint will complain.

Sorting it just maintains the order for those versions of Python where that's not standard. If we don't care about that I can remove the sorting. If we do care, something like `cached_property` might solve the performance issue here.

Sorting also made the unit tests easier to write, but that's not necessarily a reason to keep this around.